File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.php
class Solution {

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle) {
        $lh = strlen($haystack);
        $ln = strlen($needle);
        
        if ($lh == $ln){
            if ($haystack == $needle){
                return 0;
            } 
            return -1;
        }
        for($i=0;$i<=$lh-$ln;$i++){
            if (substr($haystack,$i,$ln) == $needle){
                return $i;
            }
        }
        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        let lh = haystack.len();
        let ln = needle.len();
        
        if lh < ln{
            return -1
        }
        if lh == ln{
            if haystack == needle{
                return 0;
            }
            return -1;
        }
        for i in 0..=lh-ln {
            let h = &haystack[i..i+ln];
            println!("{i}, {lh}, {ln}, {h}");
            if h == needle{
                return i as i32;
            }
        }
        return -1;
    }
}